Packs/Fortanix-DSM/Integrations/FortanixDSM/FortanixDSM.py

```diff
@@ -0,0 +1,525 @@
+            k_v = attr.strip().split('=')
+            metadata[k_v[0]] = k_v[1]
+    return metadata
+
+
+def get_headers(username, password, apikey=None, bearer=None):
+    headers = {'Content-Type': 'application/json'}
+
+    if bearer:
+        headers['Authorization'] = f'Bearer {bearer}'
+    if username and password:
+        if username.find('CERTIFICATE') < 0 and password.find('PRIVATE KEY') < 0:
+            headers['Authorization'] = 'Basic ' + \
+                b64encode(b":".join([username.encode("latin1"),
+                                     password.encode("latin1")])).decode('ascii')
+    elif apikey:
+        headers['Authorization'] = f'Basic {apikey}'
+
+    return headers
+
+
+def get_client_cert(username, password):
+    if username.find('CERTIFICATE') > 0 and password.find('PRIVATE KEY') > 0:
+        fBundle = tempfile.NamedTemporaryFile(delete=False)
+        fName = fBundle.name
+        fBundle.write(bytes(username.replace('\\n', '\n'), 'ascii'))
+        fBundle.write(bytes(password.replace('\\n', '\n'), 'ascii'))
+        fBundle.seek(0)
+        fBundle.flush()
+        fBundle.close()
+        return (fName)
+    else:
+        return None
+
+
+''' CLIENT CLASS '''
+
+
+class Client(BaseClient):
+
+    def send_request(self, path, method='get', body=None, params=None):
+        data = json.dumps(body) if body is not None else None
+        params = params if params is not None else {}
+
+        return self._http_request(method=method, url_suffix=path, data=data, params=params)
+
+
+''' COMMAND FUNCTIONS '''
+
+
+def test_module(client):
+    try:
+        path = 'sys/v1/session/auth'
+        client.send_request(path)
+        return 'ok'
+    except Exception as e:
+        raise DemistoException(f"Error in API call - check the username/password or API Key. Error: {e}.")
+
+
+def list_secrets_command(client, args, gids=None):
+    name = args.get('name')
+    kid = args.get('kid')
+    if page := args.get('page'):
+        page = int(args.get('page'))
+
+    path = 'crypto/v1/keys'
+    params = {
+        'obj_type': 'SECRET',
+        'sort': 'name:asc'
+    }
+    if state := args.get('state'):
+        if state == 'enabled':
+            state = True
+        elif state == 'disabled':
+            state = False
+        elif state == 'deleted':
+            params['show_deleted'] = 'true'
+        elif state == 'destroyed':
+            params['show_destroyed'] = 'true'
+
+    if page and page > 1:
+        params['offset'] = str((page - 1) * 100 + 1)
+    if kid:
+        path = f'{path}/{kid}'
+    elif name:
+        params['name'] = name
+    else:
+        if group_id := args.get('group_id'):
+            params['group_id'] = group_id
+        elif gids and len(gids) and gids.find(' ') < 0 and gids.find(',') < 0:
+            params['group_id'] = gids
+
+    res = client.send_request(path, 'get', params=params)
+    if not res:
+        raise DemistoException('Secrets not found')
```

Comment:
When there are no secrets found, an empty list is returned from the API? If so, it is better to return an empty list and not throw an exception. It is of if the list is empty, it should not fail the playbook or flow.

Reply:
done